components/shared/ui/menu.tsx

```diff
@@ -143,14 +146,32 @@ const DropdownMenuItems: FC<DropdownMenuProps> = (props) => {
 export const Menu: FC<Props> = (props) => {
   const [activeMenu, setActiveMenu] = useState<string | number>(-1);
   const [smallMenuActive, setSmallMenuActive] = useState(false);
+  const [isScrolled, setIsScrolled] = useState(false);
 
   const handleMenuClick = (menuId: string | number): void => {
     setActiveMenu(menuId === activeMenu ? -1 : menuId);
   };
 
+  useEffect(() => {
+    const handleScroll = () => {
+      if (window.scrollY > 50) {
+        setIsScrolled(true);
+      } else {
+        setIsScrolled(false);
+      }
+    };
+
+    window.addEventListener("scroll", handleScroll);
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
+    };
+  }, []);
+
   return (
     <div
-      className={`w-full fixed z-50 transition-all ease-in-out duration-300 ${mainColorBgClass[siteCodename]}`}
+      className={`w-full fixed z-50 transition-all ease-in-out duration-200 ${
+        (isScrolled || smallMenuActive) ? mainColorBgClass[siteCodename] :  "bg-opacity-0"

```

Comment:
the solution you propose also involves some custom JS code, but it might be possible to refactor this to some extent. however, the existing handler for menu clicking already makes use of a boolean state, so the addition of an extra state for scrolling doesn't seem to overly complicate things to me and actually makes it clear that the background is toggled by either scrolling or clicking the menu open on smaller screens